autoemulate/simulations/seir.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+import numpy as np
+import torch
+from scipy.integrate import solve_ivp
+
+from autoemulate.core.types import NumpyLike, TensorLike
+from autoemulate.simulations.base import Simulator
+
+
+def simulate_seir_epidemic(
+    x: NumpyLike,
+    N: int = 1000,
+    I0: int = 1,
+    E0: int = 0,
+) -> float:
+    """
+    Simulate an epidemic using the SEIR model.
+
+    Parameters
+    ----------
+    x : NumpyLike
+        SEIR parameters [beta, gamma, sigma].
+    N : int
+        Total population.
+    I0 : int
+        Initial infected.
+    E0 : int
+        Initial exposed.
+
+    Returns
+    -------
+    peak_infection_rate : float
+        Peak infection fraction I_max / N.
+    """
+    if len(x) != 3:
+        raise ValueError(f"Expected 3 parameters [beta, gamma, sigma], got {len(x)}")
+
+    beta, gamma, sigma = x
+
+    S0 = N - I0 - E0
+    R0 = 0
+    t_span = (0.0, 160.0)
+    y0 = [S0, E0, I0, R0]
+
+    def seir_model(t, y, N, beta, gamma, sigma):
+        S, E, I, R = y
+        dSdt = -beta * S * I / N
+        dEdt = beta * S * I / N - sigma * E
+        dIdt = sigma * E - gamma * I
+        dRdt = gamma * I
+        return [dSdt, dEdt, dIdt, dRdt]
+
+    t_eval = np.linspace(t_span[0], t_span[1], 160)
+    sol = solve_ivp(
+        seir_model,
+        t_span,
+        y0,
+        args=(N, beta, gamma, sigma),
+        t_eval=t_eval,
+        vectorized=False,
+    )
+
+    _, E, I, R = sol.y
```

Comment:
same as above (ambiguous variable names)
```suggestion
    _, E, I, R = sol.y  # noqa: E741 
```